pkg/controller/windowsmachine/nodeconfig/nodeconfig.go

```diff
@@ -46,6 +49,8 @@ type nodeConfig struct {
 	node *v1.Node
 	// network holds the network information specific to the node
 	network *network
+	// signer is used for authenticating against the VM
+	signer ssh.Signer
```

Comment:
You just need the hash here and don't need to pass `signer` unless I am missing something.